src/contexts/ProductsListContext.js

```diff
@@ -1,14 +1,12 @@
 import axios from "axios";
 import { createContext, useContext, useEffect, useReducer, useState } from "react";
-import { productReducer } from "./ReducerFunc";
+import { productReducer, addCartReducer, wishListReducer } from "./ReducerFunc";
 
 const productContext = createContext();
 
-
-
 const ProductListProvider = ({ children }) => {
 
-    const [productState, productDispatch] = useReducer(productReducer, {
+    const [productState, productDispatch] = useReducer( productReducer, {
         sort: null,
```

Comment:
can extract this into one object as initial state so that you can use it while re setting on click of clear button instead of writing again